streams.go

```diff
@@ -203,50 +254,81 @@ outer:
 			protoFilter.Encoded = filter.Encode()
 		}
 
+		// Wait for connection to be ready before attempting to subscribe
+		c.waitForReadyConnection(5 * time.Second)
+
 		res, err := c.client.SubscribeNewTxsV2(ctx, protoFilter)
 		if err != nil {
+			cancel() // Cancel this context since we're going to retry
+
+			// After too many attempts, give up
 			if attempts > 50 {
-				return fmt.Errorf("subscribing to transactions after 50 attempts: %w", err)
+				return fmt.Errorf("failed to subscribe to transactions after 50 attempts: %w", err)
+			}
+
+			// Use exponential backoff with jitter
+			sleepTime := backoff + time.Duration(rand.Int63n(int64(backoff/2)))
+			fmt.Printf("Subscription error, retrying in %v: %v\n", sleepTime, err)
```

Comment:
We should probably implement a logger instead of this. But can we live with it?

Reply:
Have gotten feedback that people don't like it, so let's rm it for now and do a logger later